TestFoundation/TestScanner.swift

```diff
@@ -1,58 +1,483 @@
+            expectEqual($0.scanString("X"), "X", "Consume the X")
+            expectTrue($0.isAtEnd, "The X was not consumed")
+        }
     }
-
-    func test_scanString() {
-        let scanner = Scanner(string: "apple sauce")
-
-        guard let firstPart = scanner.scanString("apple ") else {
-            XCTFail()
-            return
+    
+    func testUInt64() {
+        // UInt64 long sequence:
+        withScanner(for: String(format: "%llu %llu %llu 42 + 42 0 %llu", UInt64.max / 10, UInt64.max - 1, UInt64.max, UInt64.max)) {
+            expectEqual($0.scanUInt64(), UInt64.max / 10, "Order of magnitude close to max")
+            expectEqual($0.scanUInt64(), UInt64.max - 1, "One less than max")
+            expectEqual($0.scanUInt64(), UInt64.max, "Max")
+            expectEqual($0.scanUInt64(), 42 as UInt64, "Short-sized integer")
+            expectEqual($0.scanUInt64(), 42 as UInt64, "Short-sized integer, with sign, ignoring whitespace")
+            expectEqual($0.scanUInt64(), 0 as UInt64, "Zero")
+            expectEqual($0.scanUInt64(), UInt64.max, "Max again after zero (ignoring prefix whitespace without merging this with the zero)")
+        }
+        
+        // Overflow:
+        withScanner(for: "\(UInt64.max)0") {
+            expectEqual($0.scanUInt64(), UInt64.max, "Overflow")
         }
-
-        XCTAssertEqual(firstPart, "apple ")
-        XCTAssertFalse(scanner.isAtEnd)
-
-        let _ = scanner.scanString("sauce")
-        XCTAssertTrue(scanner.isAtEnd)
     }
-
-    func test_charactersToBeSkipped() {
-        let scanner = Scanner(string: "xyz  ")
-        scanner.charactersToBeSkipped = .whitespaces
-
-        let _ = scanner.scanString("xyz")
-        XCTAssertTrue(scanner.isAtEnd)
+    
+    func testInt64() {
+        // Int64 long sequence:
+        withScanner(for: String(format: "%lld %lld %lld 42 - 42 0 -1 -1 %lld %lld", Int64.max / 10, Int64.max - 1, Int64.max, Int64.min, Int64.max)) {
+            expectEqual($0.scanInt64(), Int64.max / 10, "Order of magnitude close to max")
+            expectEqual($0.scanInt64(), Int64.max - 1, "One less than max")
+            expectEqual($0.scanInt64(), Int64.max, "Max")
+            expectEqual($0.scanInt64(), 42 as Int64, "Short-sized integer")
+            expectEqual($0.scanInt64(), -42 as Int64, "Short-sized integer, with sign, ignoring whitespace")
+            expectEqual($0.scanInt64(), 0 as Int64, "Zero")
+            expectEqual($0.scanInt64(), -1 as Int64, "Minus one")
+            expectEqual($0.scanInt64(), -1 as Int64, "Minus one after whitespace")
+            expectEqual($0.scanInt64(), Int64.min, "Min")
+            expectEqual($0.scanInt64(), Int64.max, "Max again after min (no joining it with preceding min even with ignroed whitespace)")
+        }
+        
+        // Overflow:
+        withScanner(for: "\(Int64.max)0") {
+            expectEqual($0.scanInt64(), Int64.max, "Overflow")
```

Comment:
So is there anyway to detect if a number being scanned is >`Int64.max` ? Or have a flag to say dont clip to `Int64.max` ?

Reply:
It can't not clip to a Int64 value, since it returns a Int64. This is consistent with the behavior of the older methods.